### PR TITLE
added findOne, deleteOne, UpdateById, deleteAll

### PR DESCRIPTION
**Pull request by DonJuvn**

### CODE REVIEW

File: server/api/index.js [hunk 1]
```diff
@@ -23,11 +34,37 @@ class Entity {
...
+            });
+            res.status(200).json(entities);
+        } catch (err) {
+            return res.status(400).json({ message: err.errors });
+        }
+    }
 
-    async delete() {}
+    async delete(res) {
+        const deletionCriteria = { _id: 123 };
```
Comment (darkhanzhursin) [inline server/api/index.js]: _id: 123
why 123?

File: server/api/index.js [hunk 2]
```diff
@@ -12,7 +12,18 @@ class Entity {
         }
     }
 
-    async get() {}
+    async get(res) {
+        try {
+            let id = req.params.id;
+            let entity = await this.entityModel.findByPk(id);
```
Comment (darkhanzhursin) [inline server/api/index.js]: what and where is the findByPk() method?

File: server/api/index.js [hunk 3]
```diff
@@ -23,11 +34,37 @@ class Entity {
...
+            console.log(err);
+        }
+    }
+    
 
-    async deleteAll() {}
-}
+    async deleteAll(res) {
+        try {
+            const data = await this.entityModel.destroy({where: {}, truncate: true});
```
Comment (darkhanzhursin) [inline server/api/index.js]: what method is destroy? which library are u using?

**Review by darkhanzhursin — CHANGES_REQUESTED** [review]
fixes are required

File: server/api/index.js (hunk 2 above)
Comment (DonJuvn) [reply server/api/index.js]: this is similar to findById method, PK is primary key

**Review by darkhanzhursin — APPROVED** [review]

File: server/api/index.js [hunk 4]
```diff
@@ -5,14 +5,25 @@ class Entity {
...
         }
     }
 
-    async get() {}
+    async get(res) {
+        try {
+            let id = req.params.id;
+            let entity = await this.entityModel.findById(id).exec();
+            if (data)
+                return res.status(200).json({ entity })
```
Comment (eyesofdaveed) [inline server/api/index.js]: `json({data: entity})`

File: server/api/index.js [hunk 5]
```diff
@@ -5,14 +5,25 @@ class Entity {
...
+            else
+                return res.status(400).json({ success: false, error: `No such ${entity} present`, data: [] });
```
Comment (eyesofdaveed) [inline server/api/index.js]: remove `success`

Comment (eyesofdaveed) [reply server/api/index.js]: data: [],
message: ''

File: server/api/index.js [hunk 6]
```diff
@@ -5,14 +5,25 @@ class Entity {
...
         } catch (err) {
             console.log(err);
         }
     }
 
-    async get() {}
+    async get(res) {
+        try {
+            let id = req.params.id;
+            let entity = await this.entityModel.findById(id).exec();
```
Comment (eyesofdaveed) [inline server/api/index.js]: don't use `let` unless you sure that you will modify it

File: server/routes/users.js [hunk 7]
```diff
@@ -28,4 +28,13 @@ router.put('/:id', checkRole(Array(roles.at(0))), async (req, res) => {
   }
 });
 
+router.delete('/', async (req, res) => {
```
Comment (eyesofdaveed) [inline server/routes/users.js:31]: you would receive `id` from `req.query`, to delete specific entity by it's `_id`

**Review by eyesofdaveed — APPROVED** [review]

File: server/api/index.js [hunk 8]
```diff
@@ -5,14 +5,27 @@ class Entity {
...
       res.status(200).json(createdEntity);
     } catch (err) {
       console.log(err);
     }
   }
 
-  async get() {}
+  async get(req, res) {
+    try {
+      let id = req.params.id;
```
Comment (eyesofdaveed) [inline server/api/index.js:17]: const, unless you sure that you will change the value

File: server/api/index.js [hunk 9]
```diff
@@ -5,14 +5,27 @@ class Entity {
...
+      const data = await this.entityModel.findById(id).exec();
```
Comment (eyesofdaveed) [inline server/api/index.js:18]: blank line

File: server/api/index.js [hunk 10]
```diff
@@ -5,14 +5,27 @@ class Entity {
...
+      if (data) return res.status(200).json({ data: data });
```
Comment (eyesofdaveed) [inline server/api/index.js:19]: `if (!data) return res.status(400).json({
          message: 'No Data Found',
          data: [],
        });`

Comment (eyesofdaveed) [reply server/api/index.js:19]: `return res.status(200).json({ data });`

File: server/api/index.js [hunk 11]
```diff
@@ -23,11 +36,35 @@ class Entity {
     }
   }
 
-  async update() {}
+  async update(req, res) {
```
Comment (eyesofdaveed) [inline server/api/index.js:39]: Explicitly define what should be updated, .e.g: 
`const { fieldsToUpdate} = req.body;`

**Pull request closed without merging**